### DISCUSSION TITLE
How to use clair in gitlab ci?

### DISCUSSION THREAD

**Discussion by lmilbaum** [quay/clair#1856] [Q&A] [answered]
I'm building an image in a gitlab ci workflow. I'd like to run clair on the built image. 
How would I go about it?

**Comment by hdonnay** [comment, marked as answer]
Running Clair proper is probably not what you want, as you'd need to work out how to make all the layers available over HTTP. I'd suggest you take [clair-action] and attempt to adapt what the GitHub Actions [workflow] does into a Gitlab CI job, similar to how there's some [Tekton] help in that repo.

[clair-action]: https://github.com/quay/clair-action
[workflow]: https://github.com/quay/clair-action/blob/main/workflow.yaml
[Tekton]: https://github.com/quay/clair-action/blob/main/workflow.yaml

**Reply by lmilbaum** [reply]
Thanks for the advice. Apparently, Gitlab has a built in feature for containers scanning based on Trivy and Grype.
